core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -242,27 +272,27 @@ internal class CacheFallbackOperations
 
         private static void RemoveEntriesWithMatchingId(
             ISet<string> environmentAliases,
-            string identifier,
+            string accountOrUserId,
             IDictionary<AdalTokenCacheKey, AdalResultWrapper> adalCache)
         {
-            if (string.IsNullOrEmpty(identifier))
-            {
-                CoreLoggerBase.Default.Info(CoreErrorMessages.InternalErrorCacheEmptyIdentifier);
-                return;
-            }
-
             List<AdalTokenCacheKey> keysToRemove = new List<AdalTokenCacheKey>();
 
             foreach (KeyValuePair<AdalTokenCacheKey, AdalResultWrapper> kvp in adalCache)
             {
-                string environment = new Uri(kvp.Key.Authority).Host;
-                string cachedAccountId = ClientInfo.CreateFromJson(kvp.Value.RawClientInfo).ToAccountIdentifier();
+                string rawClientInfo = kvp.Value.RawClientInfo;
 
-                if (environmentAliases.Contains(environment, StringComparer.OrdinalIgnoreCase) &&
-                    string.Equals(identifier, cachedAccountId, StringComparison.OrdinalIgnoreCase))
+                if (!String.IsNullOrEmpty(rawClientInfo))
```

Comment:
Yet another place where a silent null ref can occur. We should reconsider catching all exceptions.

Reply:
perhaps at a higher level though than here? looking forward to more general discussions on this topic

---
In reply to: [219186752](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1259#discussion_r219186752) [](ancestors = 219186752)